spring-cloud-context/src/test/java/org/springframework/cloud/autoconfigure/RefreshAutoConfigurationClassPathTests.java

```diff
@@ -24,14 +24,14 @@ public void refreshEventListenerCreated() {
 		try (ConfigurableApplicationContext context = getApplicationContext(
 				Config.class)) {
 			assertFalse(context.getBeansOfType(RefreshEventListener.class).isEmpty());
-			assertFalse(context.containsBean("refeshEndpoint"));
+			assertFalse(context.containsBean("refreshEndpoint"));
 		}
 	}
 
 	private static ConfigurableApplicationContext getApplicationContext(
 			Class<?> configuration, String... properties) {
-		return new SpringApplicationBuilder(configuration).web(false)
-				.properties(properties).run();
+		return new SpringApplicationBuilder(configuration)
+				.web(false).registerShutdownHook(true).properties(properties).run();
```

Comment:
I was trying to fix the exact exceptions shown by CircleCI failing tests. I worked locally, but I see now that was a coincidence. Will undo the change.

Regarding the exceptions I cannot find the relation with my changes, and seems similar to:
https://github.com/spring-cloud/spring-cloud-config/issues/118